src/vm/instance.rs

```diff
@@ -849,10 +898,10 @@ impl<'a> Instance<'a> {
     //
     ///////////////////////////////////////////////////////
 
-    fn define_value(&mut self, addr: ConstAddressType) -> Result<()> {
+    fn define(&mut self, address: ConstAddressType) -> Result<()> {
         let v = self.pop();
-        let id = self.read_identifier(addr)?;
-        self.top_level.set(id.clone(), v.clone());
+        let id = self.read_identifier(address)?;
+        self.top_level.define(id.clone(), v.to_owned());
```

Comment:
this doesn’t need to be cloned

```suggestion
        self.top_level.define(id, v.to_owned());
```